libdnf/base/goal.cpp

```diff
@@ -1202,6 +1202,17 @@ GoalProblem Goal::Impl::add_reason_change_to_goal(
         }
     }
     for (const auto & pkg : query) {
+        if (pkg.get_reason() == reason) {
+            // pkg is already installed with correct reason
+            transaction.p_impl->add_resolve_log(
+                GoalAction::REASON_CHANGE,
+                GoalProblem::ALREADY_INSTALLED,
+                settings,
+                pkg.get_nevra(),
+                {pkg.get_nevra()},
```

Comment:
How about instead of putting the `nevra` twice in the message, we put the `reason` there.
`{pkg.get_nevra()},` -> `{libdnf::transaction::transaction_item_reason_to_string(reason)},`
and adjust log_event appropriately